src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/Interfaces/IProcessHelper.cs

```diff
@@ -56,5 +57,17 @@ internal interface IProcessHelper
         /// <param name="exitCode">return value of exitCode</param>
         /// <returns>False if process has not exited, True otherwise</returns>
         bool TryGetExitCode(Process process, out int exitCode);
+
+        /// <summary>
+        /// The wait for parent process exit async.
+        /// </summary>
+        /// <param name="parentProcessId">
+        /// The parent process id.
+        /// </param>
+        /// <param name="requestingEntity"></param>
+        /// <returns>
+        /// The <see cref="Task"/>.
+        /// </returns>
+        Task WaitForParentProcessExitAsync(int parentProcessId, string requestingEntity);
```

Comment:
There isn't a concept of `Parent` process in this API.

I am not sure if it's the right approach to dedicate an entire thread to wait for process exit. Instead we should just set a ExitedCallback and let the consumers take an action on that.

```
IProcessHelper.SetExitCallback(int processId, Action<Process> callback)

// In testhost or datacollector, use wait handles as required
```